code/modules/mob/living/simple_animal/abnormality/waw/clouded_monk.dm

```diff
@@ -0,0 +1,220 @@
+//Coded by Coxswain
+/mob/living/simple_animal/hostile/abnormality/clouded_monk
+	name = "Clouded Monk"
+	desc = "An abnormality in the form of a tall Buddhist monk wearing a kasa hat."
+	icon = 'ModularTegustation/Teguicons/32x48.dmi'
+	icon_state = "cloudedmonk"
+	icon_living = "cloudedmonk"
+	var/icon_aggro = "pretamonk"
+	icon_dead = "pretamonk"
+	maxHealth = 2500
+	health = 2500
+	rapid_melee = 2
+	ranged = TRUE
+	damage_coeff = list(BRUTE = 1.0, RED_DAMAGE = 1.2, WHITE_DAMAGE = 0.8, BLACK_DAMAGE = 0.8, PALE_DAMAGE = 1.5)
+	melee_damage_lower = 30
+	melee_damage_upper = 45
+	melee_damage_type = RED_DAMAGE
+	see_in_dark = 10
+	stat_attack = HARD_CRIT
+	speed = 4
+	move_to_delay = 4
+	threat_level = WAW_LEVEL
+	attack_sound = 'sound/abnormalities/clouded_monk/monk_attack.ogg'
+	attack_verb_continuous = "bashes"
+	attack_verb_simple = "bash"
+	can_breach = TRUE
+	start_qliphoth = 3
+	work_chances = list(
+						ABNORMALITY_WORK_INSTINCT = 0,
+						ABNORMALITY_WORK_INSIGHT = list(20, 20, 55, 55, 55),
+						ABNORMALITY_WORK_ATTACHMENT = list(20, 45, 45, 45, 45),
+						ABNORMALITY_WORK_REPRESSION = list(40, 20, 40, 40, 40)
+						)
+	work_damage_amount = 10
+	work_damage_type = WHITE_DAMAGE
+
+	ego_list = list(
+		/datum/ego_datum/weapon/amrita,
+		/datum/ego_datum/armor/amrita
+		)
+	gift_type =  /datum/ego_gifts/amrita
+	gift_message = "Anyone can become a Buddha by washing away the anguish and delusion in their heart."
+	abnormality_origin = ABNORMALITY_ORIGIN_LOBOTOMY
+
+	var/datum/looping_sound/cloudedmonk_ambience/soundloop
+	var/charging = FALSE
+	var/charge_ready = FALSE
+	var/dash_num = 25
+	var/dash_cooldown = 0
+	var/dash_cooldown_time = 6 SECONDS
+	var/list/been_hit = list() // Don't get hit twice.
+	var/deathcount
+	var/heal_amount = 250
+	var/eaten = FALSE
+	var/damage_taken
+
+//init
+/mob/living/simple_animal/hostile/abnormality/clouded_monk/Initialize()
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_MOB_DEATH, .proc/on_mob_death) // Hell
+	soundloop = new(list(src), FALSE)
+
+/mob/living/simple_animal/hostile/abnormality/clouded_monk/Destroy()
+	UnregisterSignal(SSdcs, COMSIG_GLOB_MOB_DEATH)
+	QDEL_NULL(soundloop)
+	return ..()
+
+/mob/living/simple_animal/hostile/abnormality/clouded_monk/proc/on_mob_death(datum/source, mob/living/died, gibbed) //stolen from big bird
+	SIGNAL_HANDLER
+	if(!IsContained()) // If it's breaching right now
+		return FALSE
+	if(!ishuman(died))
+		return FALSE
+	if(died.z != z)
+		return FALSE
+	if(!died.mind)
+		return FALSE
+	deathcount++
+	if(deathcount >= 10)
```

Comment:
this may be a lot, we've had an issue with this in the past with BB and MoSB iirc, so just keep that in mind.